runtime/metricsview/executor_rewrite_percent_of_totals.go

```diff
@@ -0,0 +1,111 @@
+package metricsview
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"math"
+	"math/big"
+
+	runtimev1 "github.com/rilldata/rill/proto/gen/rill/runtime/v1"
+	"github.com/rilldata/rill/runtime"
+)
+
+func (e *Executor) rewritePercentOfTotals(ctx context.Context, qry *Query, mv *runtimev1.MetricsViewSpec, sec *runtime.ResolvedSecurity) error {
+	var measures []Measure
+	var measureIndices []int
+	for i, measure := range qry.Measures {
+		if measure.Compute != nil && measure.Compute.PercentOfTotal != nil {
+			measures = append(measures, Measure{
+				Name: measure.Compute.PercentOfTotal.Measure,
+			})
+			measureIndices = append(measureIndices, i)
+		}
+	}
+
+	if len(measures) == 0 {
+		return nil
+	}
+
+	totalsQry := &Query{
+		MetricsView: qry.MetricsView,
+		Measures:    measures,
+		TimeRange:   qry.TimeRange,
+		Where:       qry.Where,
+		TimeZone:    qry.TimeZone,
+	}
+
+	e, err := NewExecutor(ctx, e.rt, e.instanceID, mv, sec, e.priority)
+	if err != nil {
+		return err
+	}
+	defer e.Close()
+
+	res, err := e.Query(ctx, totalsQry, nil)
+	if err != nil {
+		return err
+	}
+	defer res.Close()
+
+	if !res.Next() {
+		return errors.New("query returned no results")
+	}
+
+	rowMap := make(map[string]any)
+	err = res.MapScan(rowMap)
+	if err != nil {
+		return err
+	}
+
+	for i, measure := range measures {
+		t, ok := rowMap[measure.Name]
+		if !ok {
+			return fmt.Errorf("measure %q didnt return data", measure.Name)
+		}
+		tf, ok := numberLikeToFloat64(t)
+		if !ok {
+			return fmt.Errorf("%q is not a number", measure.Name)
+		}
```

Comment:
If you create a slice of floats and pass it to `Scan`, you can lean into the `database/sql` drivers ability to scan numbers into a `float64`. Similar to this logic in the Parquet exporter: https://github.com/rilldata/rill/blob/36d9921ea0f2f13ebc3b262412853c834eff13a2/runtime/drivers/file/model_executor_olap_self.go#L295

It might look something like this:
```go
vals := make([]*float64, len(measures))
for i := range vals {
  vals[i] = new(float64)
}

for res.Next() {
  err := rows.Scan(vals...)
  // ...
}
```